e2e_tests/cli/test_import.py

```diff
@@ -0,0 +1,333 @@
+import tempfile
+import zipfile
+import darwin.datatypes as dt
+from typing import List
+
+
+def validate_downloaded_annotations(
+    tmp_dir: Path, import_dir: Path, appending: bool = False
+):
+    """
+    Validate that the annotations downloaded from a release match the annotations in
+    a particular directory, ignoring hidden files.
+
+    If `appending` is set, then the number of actual annotations should exceed the
+    number of expected annotations
+    """
+    annotations_dir = tmp_dir / "annotations"
+    with zipfile.ZipFile(tmp_dir / "dataset.zip") as z:
+        z.extractall(annotations_dir)
+        expected_annotation_files = {
+            file.name: str(file)
+            for file in import_dir.iterdir()
+            if file.is_file() and not file.name.startswith(".")
+        }
+        actual_annotation_files = {
+            file.name: str(file)
+            for file in annotations_dir.iterdir()
+            if file.is_file() and not file.name.startswith(".")
+        }
+        for file in expected_annotation_files:
+            assert file in actual_annotation_files
+            expected_annotations: List[dt.Annotation] = parse_darwin_json(
+                Path(expected_annotation_files[file])
+            ).annotations  # type: ignore
+            actual_annotations: List[dt.Annotation] = parse_darwin_json(
+                Path(actual_annotation_files[file])
+            ).annotations  # type: ignore
+
+            # Delete generated UUIDs as these will break asserting equality
+            for annotation in expected_annotations:
+                del [annotation.id]  # type: ignore
+                if annotation.annotation_class.annotation_type == "raster_layer":
+                    del [annotation.data["mask_annotation_ids_mapping"]]  # type: ignore
+            for annotation in actual_annotations:
+                del [annotation.id]  # type: ignore
+                if annotation.annotation_class.annotation_type == "raster_layer":
+                    del [annotation.data["mask_annotation_ids_mapping"]]  # type: ignore
+
+            # Check that every expected annotation was imported
+            if appending:
+                assert len(actual_annotations) > len(expected_annotations)
+            else:
+                assert len(expected_annotations) == len(actual_annotations)
+
+            for expected_annotation in expected_annotations:
+                expected_annotation_data = expected_annotation.data
+                expected_annotation_type = (
+                    expected_annotation.annotation_class.annotation_type
+                )
+                actual_annotation = next(
+                    (
+                        annotation
+                        for annotation in actual_annotations
+                        if annotation.data == expected_annotation_data
+                        and annotation.annotation_class.annotation_type
+                        == expected_annotation_type
```

Comment:
This is necessary because the `data` field of both the `tag` and `mask` types is just `{}`, so we need to check that `annotation_type` matches too